src/calendar/date/CalendarUtils.ts

```diff
@@ -399,17 +410,43 @@ function getCalculationEvent(event: CalendarEvent, zone: string, handleAsAllDay:
 	}
 }
 
-//also checks whether events overlap due to minimum height
+/**
+ * This function checks whether two events collide based on their start and end time
+ * Assuming vertical columns with time going top-to-bottom, this would be true in these cases:
+ *
+ * case 1:
+ * +-----------+
+ * |           |
+ * |           |   +----------+
+ * +-----------+   |          |
+ *                 |          |
+ *                 +----------+
+ * case 2:
+ * +-----------+
+ * |           |   +----------+
+ * |           |   |          |
+ * |           |   +----------+
+ * +-----------+
+ *
+ * There could be a case where they are flipped vertically, but we don't have them because earlier events will be always first. so the "left" top edge will
+ * always be "above" the "right" top edge.
+ */
 function collidesWith(a: CalendarEvent, b: CalendarEvent): boolean {
-	return (a.endTime.getTime() > b.startTime.getTime() && a.startTime.getTime() < b.endTime.getTime()) || overlapsWith(a.startTime, a.endTime, b.startTime)
+	return (a.endTime.getTime() > b.startTime.getTime() && a.startTime.getTime() < b.endTime.getTime())
 }
 
-//Due to the minimum height for events they overlap if a short event is directly followed by another event,
-//therefore, we check whether the event height is less than the minimum height
-function overlapsWith(firstEventStart: Date, firstEventEnd: Date, secondEventStart: Date): boolean {
-	//we are only interested in the height on the last day of the event
+/**
+ * Due to the minimum height for events they overlap if a short event is directly followed by another event,
+ * therefore, we check whether the event height is less than the minimum height.
+ *
+ * This does not cover all the cases but handles the case when the second event starts right after the first one.
+ */
+function visuallyOverlaps(firstEventStart: Date, firstEventEnd: Date, secondEventStart: Date): boolean {
```

Comment:
same here

Reply:
We tried writing tests for it but decided against it since the layout computation is very unclear and probably buggy.
It adds vertical padding to all events with no obvious reason.
We plan to do more investigation in the future, but other issues have priority now.